subworkflows/local/manipulate_damage.nf

```diff
@@ -0,0 +1,134 @@
+//
+// Calculate PMD scores, trim, or rescale DNA damage from mapped reads.
+//
+
+include { MAPDAMAGE2                                       } from '../../modules/nf-core/mapdamage2/main'
+include { PMDTOOLS_FILTER                                  } from '../../modules/nf-core/pmdtools/filter/main'
+include { BAMUTIL_TRIMBAM                                  } from '../../modules/nf-core/bamutil/trimbam/main'
+include { SAMTOOLS_INDEX as SAMTOOLS_INDEX_DAMAGE_RESCALED } from '../../modules/nf-core/samtools/index/main'
+include { SAMTOOLS_INDEX as SAMTOOLS_INDEX_DAMAGE_FILTERED } from '../../modules/nf-core/samtools/index/main'
+include { SAMTOOLS_INDEX as SAMTOOLS_INDEX_DAMAGE_TRIMMED  } from '../../modules/nf-core/samtools/index/main'
+include { SAMTOOLS_FLAGSTAT as SAMTOOLS_FLAGSTAT_FILTERED  } from '../../modules/nf-core/samtools/flagstat/main'
+
+// TODO: Add required channels and channel manipulations for reference-dependent bed masking before pmdtools. Requires multi-ref support before implementation.
+workflow MANIPULATE_DAMAGE {
+    take:
+    ch_bam_bai  // [ [ meta ], bam , bai ]
+    ch_fasta    // [ [ meta ], fasta ]
+
+    main:
+    ch_versions              = Channel.empty()
+    ch_multiqc_files         = Channel.empty()
+    ch_rescaled_bams         = Channel.empty()
+    ch_pmd_filtered_bams     = Channel.empty()
+    ch_trimmed_bams          = Channel.empty()
+    ch_pmd_filtered_flagstat = Channel.empty() // Only run flagstat on pmd filtered bam, since rescaling and trimming does not change the number of reads
+
+    // Ensure correct reference is associated with each bam_bai pair
+    ch_refs = ch_fasta.map {
+        // Create additional map containing only meta.reference for combining samples and reference fastas
+        meta, fasta ->
+            meta2 = [:]
+            meta2.reference = meta.id
+        [ meta2, meta, fasta ]
+    }
+
+    ch_input_for_damage_manipulation = ch_bam_bai.map {
+        meta, bam, bai ->
+            meta2 = [:]
+            meta2.reference = meta.reference
+        [ meta2, meta, bam, bai ]
+    }
+    .combine(ch_refs, by: 0 ) // [ [combine_meta], [meta], bam, bai, [ref_meta], fasta ]
+
+    if ( params.run_mapdamage_rescaling ) {
+        ch_mapdamage_prep = ch_input_for_damage_manipulation.branch {
+            ignore_me, meta, bam, bai, ref_meta, fasta ->
+            skip:    meta.damage_treatment == 'full'
+            no_skip: true
+        }
+
+        ch_skip_rescale = ch_mapdamage_prep.skip.map {
```

Comment:
Fine to keep the subbranch on first line, but please new line before the operation (`.map`)